components/camel-aws/camel-aws2-sqs/src/main/java/org/apache/camel/component/aws2/sqs/Sqs2Consumer.java

```diff
@@ -404,14 +434,13 @@ public void cancel() {
         @Override
         public void run() {
             if (run.get()) {
-                ChangeMessageVisibilityRequest.Builder request
-                        = ChangeMessageVisibilityRequest.builder().queueUrl(getQueueUrl()).visibilityTimeout(repeatSeconds)
-                                .receiptHandle(exchange.getIn().getHeader(Sqs2Constants.RECEIPT_HANDLE, String.class));
+                ChangeMessageVisibilityBatchRequest.Builder request
+                        = ChangeMessageVisibilityBatchRequest.builder().queueUrl(getQueueUrl()).entries(entries);
```

Comment:
Please be aware that the request can take at most 10 entries accordingly to the AWS documentation. `entries` at this point could be more than that.

Reply:
Thanks then we should split the batch into group of 10, and then if there are 25, then we do 3 calls.